### PR TITLE
czmq: download url changed to github releases page

### PR DESCRIPTION
**Pull request by JayThomason**
- [x] Have you followed the guidelines in our [Contributing](https://github.com/Homebrew/homebrew-core/blob/master/.github/CONTRIBUTING.md) document?
- [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally prior to submission with `brew install <formula>` (where `<formula>` is the name of the formula you're submitting)?
- [x] Does your submission pass `brew audit --strict --online <formula>` (after doing `brew install <formula>`)?

---

Accompanying changes required due to lack of configure script in new tarball.

### CODE REVIEW

**Comment by JayThomason** [comment]
Not sure why the selftest seems to fail on mavericks.

File: Formula/czmq.rb [hunk 1]
```diff
@@ -16,14 +16,13 @@ class Czmq < Formula
 
   head do
     url "https://github.com/zeromq/czmq.git"
```
Comment (DomT4) [inline Formula/czmq.rb]: If there's nothing in the `head` block beside the `url` you can just move that line to one empty line beneath the `revision 3` line up top and rename the variable `head`.

**Comment by DomT4** [comment]
Happened on Yosemite instead of Mavericks on the rebuild as well:

```
==> make check
Making check in doc
make[1]: Nothing to be done for `check'.
/Applications/Xcode.app/Contents/Developer/usr/bin/make  src/czmq_selftest
  CC       src/src_czmq_selftest-czmq_selftest.o
  CCLD     src/czmq_selftest
/Applications/Xcode.app/Contents/Developer/usr/bin/make  check-TESTS check-local
/bin/sh ./libtool --mode=execute ./src/czmq_selftest
============================================================================
Testsuite summary for czmq 3.0.2
============================================================================
# TOTAL: 0
# PASS:  0
# SKIP:  0
# XFAIL: 0
# FAIL:  0
# XPASS: 0
# ERROR: 0
============================================================================
Assertion failed: (zpoller_wait (watch_poll, 1001) == watch), function zdir_test, file src/zdir.c, line 999.
Running czmq selftests...
 * zactor: OK
 * zauth: OK
 * zarmour: OK
 * zbeacon: OK
 * zcert: OK
 * zcertstore: OK
 * zchunk: OK
 * zclock: OK
 * zconfig: OK
 * zdigest: OK
 * zdir: make[2]: *** [check-local] Abort trap: 6
make[1]: *** [check-am] Error 2
make: *** [check-recursive] Error 1
```

Maybe a race condition or something if the failure isn't Mavericks-specific and hasn't come up before now. Unsure.

**Comment by JayThomason** [comment]
@DomT4 This time the build seemed to work.

I believe there could be a race condition below if `newfile` is not written to disk within 1001 milliseconds, which seems possible: 
```    
zsock_send (watch, "ss", "SUBSCRIBE", "zdir-test-dir");
assert (zsock_wait (watch) == 0);

zfile_t *newfile = zfile_new ("zdir-test-dir", "test_abc");
zfile_output (newfile);
fprintf (zfile_handle (newfile), "test file\n");
zfile_close (newfile);

zpoller_t *watch_poll = zpoller_new (watch, NULL);

// poll for a certain timeout before giving up and failing the test.
assert(zpoller_wait (watch_poll, 1001) == watch);
```

**Comment by JayThomason** [comment]
@DomT4 If you want to further debug the build failure we can change `make check` to `make check-verbose` and run the build a few times.

**Comment by JayThomason** [comment]
bump

**Comment by DomT4** [comment]
Apologies for the delay, life is well, life. It'd be good to dig into this a bit further because I presume if CI can occasionally run into it users can as well, and that'd be a bigger deal.

**Comment by JayThomason** [comment]
No problem. I'll try running the verbose check so maybe we can get some more specific debug info.

**Comment by DomT4** [comment]
Thanks! Appreciate the willingness to dig into this. Got a decent failure off Yosemite [this time](http://bot.brew.sh/job/Homebrew%20Core%20Pull%20Requests/2913/version=yosemite/testReport/junit/brew-test-bot/yosemite/install_czmq/).

**Comment by JayThomason** [comment]
``` 
Invalid argument (src/tcp.cpp:60)
...
 * zmsg: OK
 * zpoller: OK
 * zproxy: 
I: 16-06-08 23:17:13 zproxy: API command=FRONTEND
make: *** [check-verbose] Abort trap: 6
```
Unfortunately this is pretty difficult to track down -- looks like a call to disable TCP_NODELAY for a socket failed although I don't know why that might happen. It is unclear what is going on given that we have had multiple different tests run inconsistently on different operating systems.

**Comment by JayThomason** [comment]
I've seen this error: ```brew-test-bot.xml.[failed-to-read]``` at least twice on the mavericks build.

**Comment by DomT4** [comment]
Aye, we stepped on most of those cases but have apparently discovered a new one. @apjanke may be interested there.

**Comment by apjanke** [comment]
For future reference, here's the error message from the failed build. Relevant phrase: `An invalid XML character (Unicode: 0x1e) ...`

```
Failed to read test report file /Users/brew/Jenkins/pr-core/version/mavericks/brew-test-bot.xml
org.dom4j.DocumentException: Error on line 1812 of document file:///Users/brew/Jenkins/pr-core/version/mavericks/brew-test-bot.xml : An invalid XML character (Unicode: 0x1e) was found in the CDATA section. Nested exception: An invalid XML character (Unicode: 0x1e) was found in the CDATA section.
```

**Comment by apjanke** [comment]
I was not able to reproduce this `test-bot` breakage on my local Mavericks machine. (Doing `brew test-bot` on this PR's changes on a local branch runs without error.) But I think this PR should fix it anyway: https://github.com/Homebrew/brew/pull/341.

Since it's a patch to the testing code itself, I don't know of a good way to test that PR, aside from just merging it and rebasing this `czmq` PR on it and seeing if it works. Suggestions welcome.

**Comment by bluca** [comment]
Hi,
Note that we now have uploaded the make-dist tarballs (ie, with the configure script) that were originally on download.zeromq.org to github under the respective tags.

eg: https://github.com/zeromq/czmq/releases/download/v3.0.2/czmq-3.0.2.tar.gz

**Comment by DomT4** [comment]
Nice, thank you for that (and letting us know about it) @bluca.

@JayThomason Would you mind switching to use those instead so we can escape enforcing the autotool deps on anyone who wants to build from source? Thanks!

**Comment by apjanke** [comment]
I merged Homebrew/brew#341, which should fix the "Fail to read..." breakage here. Kicked off a rerun; we'll see if it worked.

**Comment by DomT4** [comment]
Looks like that fixed it, thanks @apjanke. Would still like us to use the release tarballs here to avoid the autotool dependencies for all from-source builders.

**Comment by JayThomason** [comment]
@DomT4 Of course, I'll do that now.

**Comment by JayThomason** [comment]
Yet another Mavericks test failure, this time: ```Assertion failed: (received_port == port_nbr), function zbeacon_v2_test, file src/zbeacon_v2.c, line 533```

Looking at the source, the only way I can see that failing would be if there were another udp service on the same broadcast port, which I suppose is entirely possible if the tests are running at the same time and the test machines are on the same local network.

I'll try forcing the test to use localhost.

**Comment by DomT4** [comment]
Merged in https://github.com/Homebrew/homebrew-core/commit/dbcccfa8fa132fe23704da114e05cb35b60012bd. Thank you for your contribution to Homebrew @JayThomason; we appreciate it! :smiley_cat:

**Comment by JayThomason** [comment]
@DomT4 Glad to help.

**Pull request merged**